cmd/node/node.go

```diff
@@ -535,9 +537,30 @@ func (app *App) initServices(ctx context.Context,
 
 	txHandler := txs.NewTxHandler(app.conState, app.addLogger(TxHandlerLogger, lg))
 
+	// TODO: we should probably decouple the apptest and the node (and duplicate as necessary) (#1926)
+	var hOracle hare.Rolacle
+	if isFixedOracle {
+		// fixed rolacle, take the provided rolacle
+		hOracle = rolacle
+	} else {
+		// regular oracle, build and use it
+		hOracle = eligibility.New(beaconProtocol, cdb, signing.VRFVerify, vrfSigner, app.Config.LayersPerEpoch, app.Config.HareEligibility, app.addLogger(HareOracleLogger, lg))
+		// TODO: genesisMinerWeight is set to app.Config.SpaceToCommit, because PoET ticks are currently hardcoded to 1
+	}
+	app.certifier = blocks.NewCertifier(sqlDB, hOracle, nodeID, sgn, app.host, clock,
+		blocks.WithCertContext(ctx),
+		blocks.WithCertConfig(blocks.CertConfig{
+			CommitteeSize:         app.Config.HARE.N,
+			CertifyThreshold:      app.Config.HARE.F + 1,
+			SignatureWaitDuration: app.Config.HARE.WakeupDelta,
+			NumLayersToKeep:       app.Config.Tortoise.Zdist,
```

Comment:
Missed this before. Why Zdist instead of Hdist?

Reply:
it can be arbitrary numbers really. just the amount of old data to keep.
tortoise waits for zdist layer to give up waiting for hare output so just to be consistent.

hdist is the number of layers tortoise will use the hare output. it's not relevant here as syncer will be taking over to sync cert from peers so that tortoise can still use the synced cert within hdist.